packages/html-reporter/src/testFileView.tsx

```diff
@@ -18,35 +18,38 @@ import type { HTMLReport, TestCaseSummary, TestFileSummary } from './types';
 import * as React from 'react';
 import { hashStringToInt, msToString } from './utils';
 import { Chip } from './chip';
-import { filterWithToken, type Filter } from './filter';
+import { filterWithToken } from './filter';
 import { generateTraceUrl, Link, navigate, ProjectLink } from './links';
 import { statusIcon } from './statusIcon';
 import './testFileView.css';
 import { video, image, trace } from './icons';
+import { useSearchParams } from './useSearchParams';
 
 export const TestFileView: React.FC<React.PropsWithChildren<{
   report: HTMLReport;
   file: TestFileSummary;
   isFileExpanded: (fileId: string) => boolean;
   setFileExpanded: (fileId: string, expanded: boolean) => void;
-  filter: Filter;
-}>> = ({ file, report, isFileExpanded, setFileExpanded, filter }) => {
+}>> = ({ file, report, isFileExpanded, setFileExpanded }) => {
+  const searchParams = useSearchParams();
+  const q = searchParams.get('q') ?? '';
+
   return <Chip
     expanded={isFileExpanded(file.fileId)}
     noInsets={true}
     setExpanded={(expanded => setFileExpanded(file.fileId, expanded))}
     header={<span>
       {file.fileName}
     </span>}>
-    {file.tests.filter(t => filter.matches(t)).map(test =>
+    {file.tests.map(test =>
       <div key={`test-${test.testId}`} className={'test-file-test test-file-test-outcome-' + test.outcome}>
         <div className='hbox' style={{ alignItems: 'flex-start' }}>
           <div className="hbox">
             <span className="test-file-test-status-icon">
               {statusIcon(test.outcome)}
             </span>
             <span>
-              <Link href={`#?testId=${test.testId}`} title={[...test.path, test.title].join(' › ')}>
+              <Link href={`#?q=${q}&testId=${test.testId}`} title={[...test.path, test.title].join(' › ')}>
```

Comment:
With the URL structure like this I can no longer share a link to a test document (?testId) w/o the query involved. That does not sound right. I'll stop my review here since it is pretty fundamental.

Reply:
I am not sure what would stop you from sharing this link. It will obviously include a filter query param but it should still work 🤔 

Remembering filters is quite important to determine the next items. Passing it as a query param seems like the most natural way. TBH I don't see any other way besides some hacky solutions or rewriting a lot of the code.

If that's a deal-breaker I will just close the PR @pavelfeldman

Reply:
Imagine me making a bookmark to a test failure and then re-opening it on the next run. But that next time query does not include the test...

Reply:
The mode that would retain the test list, while compressing it to the left and opening a wider side view with the test view would make more sense. It would not need to be URL-addressible and would allow convenient navigation. You would be able to pop out (or press copy link button) for test to get into a URL-addressible mode.

Reply:
> The mode that would retain the test list, while compressing it to the left and opening a wider side view with the test view would make more sense. It would not need to be URL-addressible and would allow convenient navigation. You would be able to pop out (or press copy link button) for test to get into a URL-addressible mode. Let me know if you want to give it a try!

@pavelfeldman:
I'd like to pick this up. Just to clarify the design you're describing here, are you meaning having something similar to this view when a test case is navigated to?

![image](https://github.com/user-attachments/assets/85bc28a0-3093-4fc8-86b6-73c9b08cc1b3)